### PR TITLE
Closes #16345: Standartizing share icon across all app

### LINKED ISSUE #16345: [Bug] Update share icon in multi-select mode
## Steps to reproduce
1. Go to History or Bookmarks
2. Long press on entry to go into multi-select mode 

### Expected behavior
Share icon looks the same across the app (should look like the one that is in the site menu)

### Actual behavior
Share icon looks different

### PR DESCRIPTION
**Pull request by LeoMaldonado**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

This PR closes #16345. I removed share icon outlined by a filled share icon and put it in all references to share icons.

Screenshot_20201112-165052
Screenshot_20201112-165153
Screenshot_20201112-165258

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 1]
```diff
@@ -125,15 +125,14 @@ class DefaultToolbarMenu(
...
-            imageResource = R.drawable.mozac_ic_share,
+            imageResource = R.drawable.ic_share_filled,
             contentDescription = context.getString(R.string.browser_menu_share),
             iconTintColorResource = primaryTextColor(),
             listener = {
                 onItemTapped.invoke(ToolbarMenu.Item.Share)
             }
         )
 
-        registerForIsBookmarkedUpdates()
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:136]: Why are we removing this?

**Review by ekager — CHANGES_REQUESTED** [review]
Could we just remove the `ic_hollow_share` altogether and use the AC `mozac_ic_share` when needed? The only drawback I can think of is we need to make sure it's tinted correctly where used.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 1 above)
Comment (LeoMaldonado) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:136]: Don't know, putting it back

**Comment by LeoMaldonado** [comment]
> Could we just remove the `ic_hollow_share` altogether and use the AC `mozac_ic_share` when needed? The only drawback I can think of is we need to make sure it's tinted correctly where used.

You're right, but `mozac_ic_share` belongs to a third party lib, and does not follow app theme. I decided this approach because I could use `primaryText` as default color, automatically following any future color changes

**Comment by ekager** [comment]
bors try

**Review by ekager — APPROVED** [review]
Okay I'm fine with changing the asset!

**Comment by ekager** [comment]
Merged, thanks for the contribution!

**Pull request merged**